drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -5943,12 +5943,22 @@ Vector<uint8_t> RenderingDeviceVulkan::buffer_get_data(RID p_buffer) {
 
 	VkCommandBuffer command_buffer = frames[frame].setup_command_buffer;
 
+	// Size of buffer to retreive.
+	if (!p_size) {
+		p_size = buffer->size;
+	} else {
+		ERR_FAIL_COND_V_MSG((p_size % 4) != 0, Vector<uint8_t>(),
+				"Size must be a multiple of four.");
+		ERR_FAIL_COND_V_MSG(p_size > buffer->size, Vector<uint8_t>(),
```

Comment:
```suggestion
		ERR_FAIL_COND_V_MSG(p_size + p_offset > buffer->size, Vector<uint8_t>(),
```
The size combined with offset needs to be smaller than the buffer size otherwise it will attempt to read from a location outside the buffer. ``vkCmdCopyBuffer`` will also throw an error in that case

Reply:
Ah yes, that slipped by mind when I added the offset.